File: examples/histogram_mt/run/run_test_histogram.rb
#!/usr/bin/env ruby

require 'anlnext'
require 'testHistogramMT'

class MyApp < ANL::ANLApp
  def setup()
    chain TestHistogramMT::CreateRootFile
    with_parameters(filename_base: "output",
                    parallel: true,
                    save_parallel: false)

    chain TestHistogramMT::GenerateEvents
    with_parameters(energy: 120.0,
                    sigma: 4.0,
                    num_detectors: 1000,
                    efficiency: 0.03,
                    random_seed: 500)

    chain TestHistogramMT::FillHistogram
    with_parameters(nbin: 128,
                    energy_min: 80.0,
                    energy_max: 150.0)
  end
end

a = MyApp.new
a.num_parallels = 4
a.modify do |m|
  m.modify_parameters 0, :GenerateEvents, {
    energy: 90.0,
    sigma: 6.0,
  }
end

a.run(1000000)

File: examples/histogram_mt/run_py/run_test_histogram.py
#!/usr/bin/env python3

import anlnext
import testHistogramMTPy as th

### chain definition
def setup(module):
    module.chain(th.CreateRootFile)
    module.with_parameters({
        "filename_base": "output",
        "parallel": True,
        "save_parallel": False
    })

    module.chain(th.GenerateEvents)
    module.with_parameters({
        "energy": 120.0,
        "sigma": 4.0,
        "num_detectors": 1000,
        "efficiency": 0.03,
        "random_seed": 500
    })

    module.chain(th.FillHistogram)
    module.with_parameters({
        "nbin": 128,
        "energy_min": 80.0,
        "energy_max": 150.0
    })


### run analysis chain
a = anlnext.AnalysisChain()
a.num_parallels = 4
setup(a)

def modify_param(module):
    module.get_parallel_module(0, "GenerateEvents").set_parameter("energy", 90.0)
    module.get_parallel_module(0, "GenerateEvents").set_parameter("sigma", 6.0)

a.modify(modify_param)

a.run(1000000)
